doc/contributing/CODING_STANDARDS.md

```diff
@@ -232,6 +232,25 @@ a null pointer.
     .def("setHost", py::arg("host").none(false))
 ```
 
+### Polymorphic override macros
+
+In order to support C++ class instances transparently calling into a
+mirrored Python instance, pybind11 has support for so-called
+["trampoline" classes](https://pybind11.readthedocs.io/en/stable/advanced/classes.html#overriding-virtual-functions-in-python),
+which dispatch either to the Python method override, or to the C++ base
+class implementation if no Python override is found. Method bodies
+use convenience macros to implement this logic.
+
+In pybind11, any exception that occurs in Python, and propagates to C++,
+is translated to an `error_already_set` C++ exception. In order to
+support Python->C++ exception type translation, we must augment the
+pybind11 macros.
+
+So instead of using the `PYBIND11_OVERRIDE_*` family of macros, we must
+use `OPENASSETIO_PYBIND11_OVERRIDE_*`, which decorates the pybind11
```

Comment:
[question] if wonder if its worth a grep test in CI to assert this? like we do for OAIO?

Reply:
Yeah good idea. I was thinking we could change the `business-language` check to `disallowed-phrases` and add it there. Could also be useful if/when we do inclusive terminology checks too.

Reply:
CI job updated in 18647406

Showing that "OAIO" is still caught: https://github.com/OpenAssetIO/OpenAssetIO/actions/runs/6876912503/job/18703496245?pr=1190

Showing that bad pybind11 macros is caught: https://github.com/OpenAssetIO/OpenAssetIO/actions/runs/6876921789/job/18703536758?pr=1190

The ID of the job has changed and its a required check, so we can either change the ID back or update the project settings.

Reply:
Awesome sauce. The new name is better so we can just update the branch protection requirement.